pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -553,6 +553,12 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyView model.ProxyView, se
 		if !instance.Endpoint.IsDiscoverableFromProxy(&model.Proxy{Metadata: &model.NodeMetadata{ClusterID: istio_cluster.ID(cb.clusterID)}}) {
 			continue
 		}
+
+		// TODO(stevenctl) share code with EDS to filter this and do multi-network mapping
+		if instance.Endpoint.Address == "" {
```

Comment:
cc @hzxuzhonghu and @howardjohn 

I'm thinking I may completely exclude it from serviceinstances. 
Is there a chance if I do that, some kind of check like `len(svcinstances) == 0` skips building the cluster, when we would end up having shards? A quick look didn't show anything for me.

Reply:
can we run it here? 

This is for DNS cluster

Reply:
> can we run it here

I'm not sure what "it" is. Do you meant he not empty check? Or the filters from EDS? Or using endpoint shards? 

If we have a ServiceEntry like:

```
resolution: DNS
endpoints:
- address: 1.2.3.4
  network: network-2
- address: foo.bar.com
```

I think we should still translate 1.2.3.4 to use the gateway and run the same filters with EDS. Since it's not supported today, I won't try to fix it in this PR but I'll look into it later. For now I'm going to completely exclude empty addresses from service instance meaning these empty workloads will only affect EDS clusters.

Reply:
The example should not be allowed